File: Core/src/main/java/me/mrkirby153/kcutils/jukebox/SongStartedEvent.java
package me.mrkirby153.kcutils.jukebox;


import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class SongStartedEvent extends Event {

    private static final HandlerList handlers = new HandlerList();

    private NoteBlockSong song;
    private Player player;

    public SongStartedEvent(NoteBlockSong song, Player player) {
        this.song = song;
        this.player = player;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }

    @Override
    public HandlerList getHandlers() {
        return handlers;
    }

    /**
     * The player who started listening
     *
     * @return The player
     */
    public Player getPlayer() {
        return player;
    }

    /**
     * The song that is playing
     *
     * @return The Song
     */
    public NoteBlockSong getSong() {
        return song;
    }
}

File: Bukkit/src/main/kotlin/me/mrkirby153/kcutils/jukebox/SongStartedEvent.kt
package me.mrkirby153.kcutils.jukebox


import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class SongStartedEvent(
        /**
         * The song that is playing
         *
         * @return The Song
         */
        val song: NoteBlockSong,
        /**
         * The player who started listening
         *
         * @return The player
         */
        val player: Player) : Event() {

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {

        val handlerList = HandlerList()
    }
}